api_tests/nodes/views/test_node_detail.py

```diff
@@ -25,1032 +18,1151 @@
-        assert_equal(res.json['data']['attributes']['category'], self.private_project.category)
-        assert_items_equal(res.json['data']['attributes']['current_user_permissions'], self.admin_permissions)
-
-    def test_return_private_project_details_logged_in_write_contributor(self):
-        self.private_project.add_contributor(contributor=self.user_two, auth=Auth(self.user), save=True)
-        res = self.app.get(self.private_url, auth=self.user_two.auth)
-        assert_equal(res.status_code, 200)
-        assert_equal(res.content_type, 'application/vnd.api+json')
-        assert_equal(res.json['data']['attributes']['title'], self.private_project.title)
-        assert_equal(res.json['data']['attributes']['description'], self.private_project.description)
-        assert_equal(res.json['data']['attributes']['category'], self.private_project.category)
-        assert_items_equal(res.json['data']['attributes']['current_user_permissions'], self.write_permissions)
-
-    def test_return_private_project_details_logged_in_non_contributor(self):
-        res = self.app.get(self.private_url, auth=self.user_two.auth, expect_errors=True)
-        assert_equal(res.status_code, 403)
-        assert_in('detail', res.json['errors'][0])
-
-    def test_top_level_project_has_no_parent(self):
-        res = self.app.get(self.public_url)
-        assert_equal(res.status_code, 200)
-        assert_not_in('parent', res.json['data']['relationships'].keys())
-        assert_equal(res.content_type, 'application/vnd.api+json')
-
-    def test_child_project_has_parent(self):
-        public_component = NodeFactory(parent=self.public_project, creator=self.user, is_public=True)
+@pytest.fixture()
+def user():
+    return AuthUserFactory()
+
+@pytest.mark.django_db
+class TestNodeDetail:
+
+    @pytest.fixture()
+    def user_two(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def project_public(self, user):
+        return ProjectFactory(title='Project One', is_public=True, creator=user)
+
+    @pytest.fixture()
+    def project_private(self, user):
+        return ProjectFactory(title='Project Two', is_public=False, creator=user)
+
+    @pytest.fixture()
+    def component_public(self, user, project_public):
+        return NodeFactory(parent=project_public, creator=user, is_public=True)
+
+    @pytest.fixture()
+    def url_public(self, project_public):
+        return '/{}nodes/{}/'.format(API_BASE, project_public._id)
+
+    @pytest.fixture()
+    def url_private(self, project_private):
+        return '/{}nodes/{}/'.format(API_BASE, project_private._id)
+
+    @pytest.fixture()
+    def url_component_public(self, component_public):
+        return '/{}nodes/{}/'.format(API_BASE, component_public._id)
+
+    @pytest.fixture()
+    def permissions_read(self):
+        return ['read']
+
+    @pytest.fixture()
+    def permissions_write(self):
+        return ['read', 'write']
+
+    @pytest.fixture()
+    def permissions_admin(self):
```

Comment:
I think I prefer top of the file then as defaults, though might let @sloria weigh in before changing.